### PR TITLE
Add Java Maven test

### PR DESCRIPTION
**Pull request by Katka92**
### What does this PR do?
Add test for default Java Maven stack.

Once this is merged, change nightly job to run `test-all-devfiles` suite instead of `test-java-vertx`.

### CODE REVIEW

**Review by ibuziuk — APPROVED** [review]
huge +1 for getting this merged and get rid of rh-che specific tests

**Comment by dmytro-ndp** [comment]
@Katka92: is it true that this PR will make it possible to run several tests including Vert.x by one command `test-all-devfiles`?

**Comment by dmytro-ndp** [comment]
@Katka92: do we really need separate test **Login.ts** if we have to login at start of each test anyway?

**Comment by Katka92** [comment]
@dmytro-ndp 
Add 1: Yes, this command will log in and then run all tests in tests/devfiles folder. 
Add 2: But we don't have to login at the start of each test. Actually, the login at the beginning of each test is a problem because we are not able to separate this logic (login, when we are already logged, is a culprit for a failing test). In this way, the usage of login before test is optional so tests can be combined.

**Comment by dmytro-ndp** [comment]
@Katka92: thanks answers.
Let's take changes from master and fix .infra/crw-ci/ Jenkinsfile's as well: replace `test-java-vertx` with `test-all-devfiles`.

File: tests/e2e/mocha-java-vertx.opts [hunk 1]
```diff
@@ -3,5 +3,6 @@
 -u tdd 
 --bail
 --full-trace
---spec dist/tests/e2e/JavaVertx.spec.js
```
Comment (Ohrimenko1988) [inline tests/e2e/mocha-java-vertx.opts:6]: I have a suggestion about this approach, but it may be changed in the future.

**Review by Ohrimenko1988 — APPROVED** [review]

File: tests/e2e/mocha-java-vertx.opts (hunk 1 above)
Comment (Katka92) [reply tests/e2e/mocha-java-vertx.opts:6]: @Ohrimenko1988 And what the suggestion is? :smiley:

**Comment by dmytro-ndp** [comment]
@Katka92: nightly tests have passed: https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/che-regular-tests/view/nightly/job/MultiUser-Che-nightly-check-e2e-tests-against-k8s/67/
https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/basic-MultiUser-Che-check-e2e-tests-against-k8s/371/consoleFull
- Java Maven test 
- Java Vert.x test

Let's fix master and pre-release Jenkinsfile's as well.

**Review by dmytro-ndp — APPROVED** [review]
Looks good.

**Comment by Katka92** [comment]
All three Jenkins files are changed. Merging.

**Pull request merged**